ggshield/core/config/config.py

```diff
@@ -51,17 +67,27 @@ def config_path(self) -> Path:
 
     @property
     def instance_name(self) -> str:
+        return self.get_instance()[0]
+
+    @property
+    def instance_source(self) -> str:
+        return self.get_instance()[1].value
+
+    def get_instance(self) -> Tuple[str, ConfigSource]:
```

Comment:
I think we don't need the `instance_source` property and I would rename `get_instance()` to `get_instance_name_and_source()`.

Reply:
I think it is clearer to have the `instance_source` property, so that the syntax is similar to that of the api key, wdyt ?

Reply:
I am not fond of the existing `Config.instance` property: I think it's wrong to have properties which do complicated work and can raise exceptions. It's already there so I haven't changed it, but I'd rather avoid spreading the problem. Especially for a property which is only going to be called once.

Reply:
(should have explained my reasons for not liking properties)

A property is meant to represent a computed value that can be used the same way one uses a class member variable. The assumption is that it's cheap and reliable to use it, just like it's cheap and reliable to read a member variable.

Having properties do complex work breaks that assumption.

Reply:
Thanks for the clarification :)